include/open62541/server.h

```diff
@@ -1285,6 +1285,114 @@ UA_Server_triggerEvent(UA_Server *server, const UA_NodeId eventNodeId, const UA_
 
 #endif /* UA_ENABLE_SUBSCRIPTIONS_EVENTS */
 
+#ifdef UA_ENABLE_SUBSCRIPTIONS_ALARMS_CONDITIONS
+typedef enum UA_TwoStateVariableCallbackType {
+  UA_ENTERING_ENABLEDSTATE,
+  UA_ENTERING_ACKEDSTATE,
+  UA_ENTERING_CONFIRMEDSTATE,
+  UA_ENTERING_ACTIVESTATE
+} UA_TwoStateVariableCallbackType;
+
+/**
+ * callback prototype to set user specific callbacks
+ */
+typedef UA_StatusCode
+(*UA_TwoStateVariableChangeCallback)(UA_Server *server, const UA_NodeId *condition);
```

Comment:
Is there an example where the UA_TwoStateVariableChangeCallback is used?
Wouldn't I need the `leaving` callbacktypes in addition to the `entering` ones?

Reply:
The idea behind that callback is to give the user the possibilty to set additional Condition Fields, which are not set by default when Condition Fields from Type TwoStateVariableType or ConditionVariableType change. For example, when enabling a Condition, UA Spec. doesn't specify what happens to the Retain Field. It could be set to true or stay false. Of course we could set Retain Field to true separatly and then trigger an Event manually. But we will end up with two Events instead of one. The same for AcknowledgeableState, the user my want to change the severity or deactivate the alarm when the AcknowledgeableState changes to true.
So It wasn't meant to change something on entering and also on leaving the callback.

It's just a proposal based on my understanding, so feel free to correct me, if I got it wrong:)